File: Day - 1/Standard Deviation/try.c
#include <stdio.h>
#include <math.h>

void stdDev(int arr[], int n) {
    // Print your answers to 1 decimal place within this function
    double mean = 0;

    for (int i = 0; i < n; i++) {
        mean += arr[i];
    }
    mean /= n;

    double sumSquares = 0;
    for (int i = 0; i < n; i++) {
        sumSquares += pow(arr[i] - mean, 2);
    }

    double result = sqrt(sumSquares / n);

    printf("%.1f\n", result);
}

int main() {
    int n;
    scanf("%d", &n);

    int vals[n];

    for (int i = 0; i < n; i++) {
        scanf("%d", &vals[i]);
    }

    stdDev(vals, n);

    return 0;
}

File: Day - 1/Standard Deviation/try.java
import java.util.Scanner;

public class StandardDeviation {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int[] vals = new int[n];

        for (int i = 0; i < n; i++) {
            vals[i] = scanner.nextInt();
        }

        stdDev(vals);
    }

    public static void stdDev(int[] arr) {
        // Print your answers to 1 decimal place within this function
        double mean = 0;
        for (int value : arr) {
            mean += value;
        }
        mean /= arr.length;

        double sumSquares = 0;
        for (int value : arr) {
            sumSquares += Math.pow(value - mean, 2);
        }

        double result = Math.sqrt(sumSquares / arr.length);

        System.out.printf("%.1f%n", result);
    }
}
